src/MySqlConnector/Protocol/Payloads/OkPayload.cs

```diff
@@ -79,7 +75,28 @@ internal sealed class OkPayload
 						switch (kind)
 						{
 							case SessionTrackKind.Schema:
-								newSchema = Encoding.UTF8.GetString(reader.ReadLengthEncodedByteString());
+								context.Database = Encoding.UTF8.GetString(reader.ReadLengthEncodedByteString());
+								break;
+
+							case SessionTrackKind.SystemVariables:
+								var systemVariableOffset = reader.Offset + dataLength;
+								do
+								{
+									var variableSv = Encoding.ASCII.GetString(reader.ReadLengthEncodedByteString());
```

Comment:
This is potentially a lot of string allocations for server variables we don't even read. We can perform a string comparison on the UTF-8 data by using `ReadOnlySpan<byte>` and `""u8` string literals: https://github.com/mysql-net/MySqlConnector/pull/1497/commits/3326a0ec06a8b32776271a7c4de103e8088b422a.

Reply:
good point